src/services/storageService.ts

```diff
@@ -24,6 +33,45 @@ class StorageService {
             })
     }
 
+    listFiles = async (): Promise<File[]> => {
+        return supabaseClient
+            .storage
+            .from(BUCKET_NAME)
+            .list()
+            .then(resp => {
+                if (resp.error) {
+                    throw new Error(resp.error.message)
+                }
+                return resp.data?.filter(f => f.name != '.emptyFolderPlaceholder')
+            })
+    }
+
+    removeFile = async (fileName: string): Promise<any> => {
+        return supabaseClient
+            .storage
+            .from(BUCKET_NAME)
+            .remove([fileName])
+            .then(resp => {
+                if (resp.error) {
+                    throw new Error(resp.error.message)
+                }
+                return resp.data
+            })
+    }
+
+    uploadFile = async (file: any): Promise<any> => {
```

Comment:
This probably should be typed as File

Reply:
the file object format while uploading is different from file object stored in storage
so I could not use same File type for it